jxl/src/image/data_type.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) the JPEG XL Project Authors. All rights reserved.
+//
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+use std::fmt::Debug;
+
+mod private {
+    pub trait Sealed {}
+}
+
+#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+pub enum DataTypeTag {
```

Comment:
maybe add a comment here that for any new data type tag the ImageDataType  trait safety requirements must be asserted below.